fbpic/print_utils.py

```diff
@@ -0,0 +1,234 @@
+        # Print version of FBPIC
+        message = '\nFBPIC (fbpic-%s)\n'%__version__
+        # Basic information
+        if level == 1:
+            # Print information about computational setup
+            if sim.use_cuda:
+                message += "\nRunning on GPU "
+            else:
+                message += "\nRunning on CPU "
+            if sim.comm.size > 1:
+                message += "with %d MPI processes " %sim.comm.size
+            if sim.use_threading and not sim.use_cuda:
+                message += "(%d threads per process) " \
+                    %numba.config.NUMBA_NUM_THREADS
+        # Detailed information
+        if level == 2:
+            if mpi_installed:
+                message += '\nMPI available: Yes'
+                message += '\nMPI processes: %d' %sim.comm.size
+            else:
+                message += '\nMPI available: No'
+            if cuda_installed:
+                message += '\nCUDA available: Yes'
+            else:
+                message += '\nCUDA available: No'
+            if sim.use_cuda:
+                message += '\nCompute architecture: GPU (CUDA)'
+            else:
+                message += '\nCompute architecture: CPU'
+                if threading_enabled:
+                    message += '\nCPU multi-threading enabled: Yes'
+                    message += '\nThreads: %s' \
+                        %numba.config.NUMBA_NUM_THREADS
+                else:
+                    message += '\nCPU multi-threading enabled: No'
+                if mkl_installed:
+                    message += '\nFFT library: MKL'
+                else:
+                    message += '\nFFT library: pyFFTW'
+
+            message += '\n'
+            if sim.fld.n_order == -1:
+                message += '\nPSAOTD stencil order: infinite'
+            else:
+                message += '\nPSAOTD stencil order: %d' \
+                    %sim.fld.n_order
+            message += '\nParticle shape: %s' %sim.particle_shape
+            message += '\nLongitudinal boundaries: %s' %sim.comm.boundaries
+            message += '\nTransverse boundaries: reflective'
+            message += '\nGuard region size: %d ' \
+                %sim.comm.n_guard + 'cells'
+            message += '\nDamping region size: %d ' \
+                %sim.comm.n_damp + 'cells'
+            message += '\nParticle exchange period: every %d ' \
+                %sim.comm.exchange_period + 'step'
+            if sim.gamma_boost is not None:
+                message += '\nBoosted frame: Yes'
+                message += '\nBoosted frame gamma: %d' \
+                    %sim.comm.gamma_boost
+                if sim.use_galilean:
+                    message += '\nGalilean frame: Yes'
+                else:
+                    message += '\nGalilean frame: No'
+            else:
+                message += '\nBoosted frame: False'
+        message += '\n'
+        print( message )
+    if level == 2:
```

Comment:
I think I would put this as level 1 ; but I do not have a very strong feeling on this: up to you.

Reply:
I would keep it at level 2. for level 1, it should be sufficient for the user to know that the simulation is done on GPU with N number of MPI processes. (This is already printed in the case of level 1). Also this then gives consistent printing behaviour for either CPU or GPU simulations (in case of level = 1)